src/libraries/Microsoft.PowerFx.Core/Functions/IResourceManager.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core.Functions
+{
+    /// <summary>
+    /// Interface for resource manager.
+    /// </summary>
+    public interface IResourceManager
+    {
+        /// <summary>
+        /// Create a new resource element for string.
+        /// </summary>
+        /// <param name="str">String.</param>
+        /// <param name="fileType">Associated FileType.</param>
+        /// <returns>Index of the new element.</returns>
+        int NewElementFromString(string str, FileType fileType = FileType.Any);
+
+        /// <summary>
+        /// Create a new resource element for a base64 encoded string.
+        /// </summary>
+        /// <param name="base64str">Base64 encoded string.</param>
+        /// <param name="fileType">Associated FileType.</param>
+        /// <returns>Index of the new element.</returns>
+        int NewElementFromBase64String(string base64str, FileType fileType = FileType.Any);
```

Comment:
>NewElementFromBase64String [](http://example.com/codeflow?start=12&length=26)

Interfaces should be minimal. 
NewElementFromString and NewElementFromBase64String are redundant. 
Can we have 1 "new" method, and then the others are extension methods?